### PR TITLE
Added new methods for Last Refresh status field reading in UI and DB added verifications in test. Added new tests for Middleware Datasources.

### PR DESCRIPTION
**Pull request by hhovsepy**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 711ae1bd3a9e3621c870082ec3c5f59c0f53a035:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 44b280ef938810a738a402ac2a2c045fdb76cce5:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/middleware/provider.py [hunk 1]
```diff
@@ -125,3 +127,19 @@ def num_datasource_ui(self, reload_data=True):
         if reload_data:
             self.summary.reload()
         return self.summary.relationships.middleware_datasources.value
+
+    @variable(alias='ui')
+    def is_refreshed(self, reload_data=True):
+        if reload_data:
+            self.summary.reload()
+        if re.match('Success.*Minute.*Ago', self.summary.status.last_refresh.value):
```
Comment (jkandasa) [inline cfme/middleware/provider.py]: @hhovsepy As we know this is `text` we can use `self.summary.status.last_refresh.text_value`

Comment (hhovsepy) [inline cfme/middleware/provider.py]: yes, thanks for catch

Comment (mfalesni) [inline cfme/middleware/provider.py]: `value` tries to convert the string value to an appropriate type, currently int/float. If any of these conversion steps fail, the original string is returned.

**Comment by dajoRH** [comment]
Lint report for commit f60f7c2b86892d1032fbadf56c4f0330c25fe6bb:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 60d064eefe09a9e1e8fe5710606813035a3f0a4c:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b1fc7da68519cdbc2f50b4765ee9deeaf1a957e2:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8fb7992b2c00bd7ab4bf71183158f232daa6e5f6:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**